packages/core/src/telemetry/uiTelemetry.ts

```diff
@@ -209,6 +217,22 @@ export class UiTelemetryService extends EventEmitter {
       tools.totalDecisions[event.decision]++;
       toolStats.decisions[event.decision]++;
     }
+
+    // Aggregate line count data from metadata
+    if (event.metadata) {
+      if (
+        typeof event.metadata.ai_added_lines === 'number' &&
```

Comment:
sorry for the confusion. What you had previously was better than this. I was suggesting that if it is known this is a numberor undefined just check for !== undefined as that is clearer.